meta-leda-components/recipes-sdv/eclipse-kuksa/databroker-cli_0.17.0.bb

```diff
@@ -15,218 +15,204 @@
 #
 inherit cargo
 
+# If this is git based prefer versioned ones if they exist
+# DEFAULT_PREFERENCE = "-1"
 
-SRCREV_FORMAT = "cli_jsoncons_jwtcpp_turtle"
-
-SRC_URI += "git://github.com/eclipse/kuksa.val;branch=master;name=cli;protocol=https;nobranch=1 \
-            git://github.com/danielaparker/jsoncons.git;protocol=https;name=jsoncons;branch=master;destsuffix=git/kuksa-val-server/3rd-party-libs/jsoncons \
-            git://github.com/Thalhammer/jwt-cpp.git;protocol=https;name=jwtcpp;branch=master;destsuffix=git/kuksa-val-server/3rd-party-libs/jwt-cpp \
-            git://github.com/mat007/turtle.git;protocol=https;name=turtle;branch=master;destsuffix=git/kuksa-val-server/3rd-party-libs/turtle \
-            "
-SRCREV_cli = "590198a35de7b2201bdd913750157bb9778a5214"
-SRCREV_jsoncons = "af61925bb960df55331d3b6ec198042a1b133694"
-SRCREV_jwtcpp = "34bb0644ea613cfcbc09c148db9de8aa6c5612b5"
-SRCREV_turtle = "bbe01e6d9d21ff7075aba782434185a8339d44dd"
-
+SRC_URI += "gitsm://github.com/eclipse/kuksa.val.git;protocol=https;nobranch=1;branch=master"
```

Comment:
gitsm used

Reply:
What's the problem with gitsm?
Most of those recipes are generated with cargo-bitbake. Splitting the SRC_URI would make everything more complicated.

Reply:
Using separate git-fetchers for all submodules is more verbose but since we want to do IP auditing with eclipse oniro, we had to take  out all gitsm fetchers from the recipes, because their automated pipeline cannot perform as well when there are many gitsm fetchers. I assume that git submodules are not as easy to traverse.

Reply:
We got multiple options here:
1. change our recipes, which is additional effort for us, since it's generated files. Each time, we need to regenerate and re-do the manual work
2. add a feature to cargo-bitbake to generate non-gitsm SRC_URIs
3. add a feature to aliens4friends to support gitsm protocol

I think option 3) would be the best choice, as gitsm is a supported protocol by BitBake.